src/components/radio-button/common/radio-button-common.ts

```diff
@@ -0,0 +1,157 @@
+
+    /**
+     * Whether the radio button is disabled.
+     */
+    @property({ reflect: true, type: Boolean })
+    public set disabled(value: boolean) {
+      this._disabled = value;
+    }
+    public get disabled(): boolean {
+      return this._disabled || (this.group?.disabled ?? false);
+    }
+    private _disabled = false;
+
+    /**
+     * Whether the radio button is required.
+     */
+    @property({ reflect: true, type: Boolean })
+    public set required(value: boolean) {
+      this._required = value;
+    }
+    public get required(): boolean {
+      return this._required || (this.group?.required ?? false);
+    }
+    private _required = false;
+
+    /**
+     * Reference to the connected radio button group.
+     */
+    public get group(): SbbRadioButtonGroupElement | null {
+      return this._group;
+    }
+    private _group: SbbRadioButtonGroupElement | null = null;
+
+    /**
+     * Whether the radio button is checked.
+     */
+    @property({ reflect: true, type: Boolean }) public checked = false;
+
+    private _abort = new SbbConnectedAbortController(this);
+    private _handlerRepository = new HandlerRepository(this, formElementHandlerAspect);
+
+    public select(): void {
+      if (this.disabled) {
+        return;
+      }
+
+      if (this.allowEmptySelection) {
+        this.checked = !this.checked;
+      } else if (!this.checked) {
+        this.checked = true;
+      }
+    }
+
+    public override connectedCallback(): void {
+      super.connectedCallback();
+      this._group = this.closest('sbb-radio-button-group') as SbbRadioButtonGroupElement;
+
+      const signal = this._abort.signal;
+      this.addEventListener('click', (e) => this._handleClick(e), { signal });
+      this.addEventListener('keydown', (e) => this._handleKeyDown(e), { signal });
+      this._handlerRepository.connect();
+    }
+
+    public override disconnectedCallback(): void {
+      super.disconnectedCallback();
+      this._handlerRepository.disconnect();
+    }
+
+    protected override willUpdate(changedProperties: PropertyValues<this>): void {
+      if (changedProperties.has('checked')) {
+        this.handleCheckedChange(this.checked, changedProperties.get('checked')!);
+      }
+      if (changedProperties.has('disabled')) {
+        this.handleDisabledChange(this.disabled, changedProperties.get('disabled')!);
+      }
+      if (changedProperties.has('required')) {
+        this.setAttribute('aria-required', `${this.required}`);
+      }
+    }
+
+    private _handleClick(event: Event): void {
+      event.preventDefault();
+      this.select();
+    }
+
+    private _handleKeyDown(evt: KeyboardEvent): void {
+      if (evt.code === 'Space') {
+        this.select();
+      }
+    }
+
+    protected abstract handleCheckedChange(currentValue: boolean, previousValue: boolean): void;
+    protected abstract handleDisabledChange(currentValue: boolean, previousValue: boolean): void;
```

Comment:
For better visibility, I would like to have those lines right after the private props (_handlerRepository).